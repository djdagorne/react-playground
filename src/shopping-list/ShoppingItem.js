import React from 'react'

export default function ShoppingItem(props){
    ShoppingItem.defaultProps = {
        item: {
            name: null,
            checked: null
        }
    }

    return (
        <li>
            <h2 style={{textDecoration: props.item.checked ? 'line-through' : null}}>
                {props.item.name}
            </h2>
            <button type='button' onClick={() => props.onCheckItem(props.item)}>check</button>
            <button type='button' onClick={() => props.onDeleteItem(props.item)}>delete</button>
        </li>
    )
}