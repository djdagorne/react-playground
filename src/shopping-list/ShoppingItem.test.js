import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ShoppingItem from './ShoppingItem';

describe('ShoppingItem Component ', () => {
    it('renders empty given no item', () => {
        const wrapper=shallow(<ShoppingItem/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the item when supplied', () => {
        // We're adding props to the ShoppingItem for different test cases
        const testItem = { name: 'test-item', checked: false } //passing in item in proper format
        const wrapper = shallow(<ShoppingItem item={testItem} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('strikes through checked items', () => {
        const testItem = { name: 'checked-item', checked: true }//formatted
        const wrapper = shallow(<ShoppingItem item={testItem} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})