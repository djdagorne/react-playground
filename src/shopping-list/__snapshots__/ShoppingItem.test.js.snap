// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingItem Component  renders the item when supplied 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": null,
      }
    }
  >
    test-item
  </h2>
  <button
    onClick={[Function]}
    type="button"
  >
    check
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
</li>
`;

exports[`ShoppingItem Component  strikes through checked items 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": "line-through",
      }
    }
  >
    checked-item
  </h2>
  <button
    onClick={[Function]}
    type="button"
  >
    check
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
</li>
`;
